extension/src/setup/index.ts

```diff
@@ -455,6 +466,42 @@ export class Setup
     )
   }
 
+  private watchDotFolderForChanges() {
+    const cwd = getFirstWorkspaceFolder()
+
+    if (this.dotFolderWatcher || !cwd) {
+      return
+    }
+
+    const disposer = Disposable.fn()
+    this.dotFolderWatcher = disposer
+    this.dispose.track(this.dotFolderWatcher)
+
+    return createFileSystemWatcher(
+      disposable => disposer.track(disposable),
+      getRelativePattern(cwd, '**'),
+      path => this.dotFolderListener(disposer, path)
+    )
+  }
+
+  private dotFolderListener(disposer: Disposer, path: string) {
+    if (
+      !(
+        (path && (path.endsWith(gitPath.DOT_GIT) || path.includes(DOT_DVC))) ||
+        disposer.disposed
+      )
+    ) {
+      return
+    }
+
+    if (path.includes(DOT_DVC)) {
```

Comment:
[F] If the path includes `.dvc` then assume there is a project and dispose of the listener.